### ISSUE TITLE
Memory leak

### ISSUE THREAD

**Issue by Strubbl** [dunst-project/dunst#538]
I had run my PC incl. dunst for about a weekend. After the weekend it was using about 17 GB of memory. Unfortunately the PC is currently not running anymore. 

How can i help to find the memory leak?

### Installation info

- Version: `Dunst - A customizable and lightweight notification-daemon 1.3.2 (2018-05-06)` and also until at least 896c008def638d151517e02240f01d1198df09db
- Install type: `pacman -S dunst`
- Distro and version: `Arch Linux`

**Comment by Strubbl** [comment]
History length seems to be ok:
```
$ grep history_length ~/.config/dunst/dunstrc 
    history_length = 50
```

I kept valgrind running for about 8 hours and it reached over 20 GB again. Here is the valgrind log of that session:
https://gist.github.com/Strubbl/13f1d3f4799fde853b62eb8d7746c421

I guess the process is getting this large memory footprint when i am away from the computer, which means i am not reading the dunst messages and they are kept.

**Comment by bebehei** [comment]
@Strubbl Thanks for the output. Is this really everything?!? The valgrind log looks like it got truncated.

**Comment by WhyNotHugo** [comment]
@Strubbl There's a link on top of the gist to the non-truncated version.

**Comment by bebehei** [comment]
Oh sorry. Github gists don't display the whole file :see_no_evil: 

Could you please paste your configuration, too?

**Comment by Strubbl** [comment]
yes, my config is this one: https://gist.github.com/Strubbl/375e2551c1d82dcdba659995997dcf5d

**Comment by tsipinakis** [comment]
Ha! That's something I wouldn't have seen for a while since debian is still on 2.40.20.

I assume it's `librsvg`'s issue if downgrading fixes it. Best thing we could do now is try to make a test case and see how it leaks. @bebehei  since you have the problematic versions in  your package tracker (and compiling libraries is a pain) can you try out variations of the following program and see if it starts leaking?

```
```

**Comment by bebehei** [comment]
Created an [upstream bugreport](https://gitlab.gnome.org/GNOME/librsvg/issues/325) and also [a reproducer](https://gitlab.gnome.org/bebehei/memleak-testproj).

**Comment by Strubbl** [comment]
Thank you very much for your investigation.

**Comment by tsipinakis** [comment]
This should be solved with librsvg 2.44.1.

**Comment by Strubbl** [comment]
the upstream bug was reopened and it should really be solved with 2.44.2 now
